src/main/java/dev/mccue/resolve/graph/DependencyTree.java

```diff
@@ -1,20 +1,101 @@
 package dev.mccue.resolve.graph;
 
 import dev.mccue.resolve.core.Dependency;
-import dev.mccue.resolve.doc.Coursier;
-import dev.mccue.resolve.doc.Incomplete;
 
-import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
 
-@Incomplete
-@Coursier("https://github.com/coursier/coursier/blob/f5f0870/modules/core/shared/src/main/scala/coursier/graph/DependencyTree.scala")
-public interface DependencyTree {
-    Dependency dependency();
+public sealed interface DependencyTree {
 
-    boolean excluded();
+    //TODO roots0 can be optionally null
+    static List<DependencyTree> of(
+            Resolution resolution,
+            List<Dependency> roots,
+            boolean withExclusions
+    ) {
+        //TODO Delete after you write code
+        record Person(String name, int age) {
+        }
 
-    String reconciledVersion();
-    String retainedVersion();
+        List<Person> persons = new ArrayList<>();
+        persons.sort(Comparator.comparing(Person::age));
+        persons.sort(Comparator.nullsFirst(Comparator.comparing(Person::name)).thenComparing(Comparator.comparing(Person::age)));
 
-    List<DependencyTree> children();
+        Set<Integer> ages = Set.of();
+
+        persons.stream()
+                .sorted(Comparator.comparing(Person::age))
+                .filter(person -> !ages.contains(person.age))
+                .map(person -> new Object[]{person.age})
+                .toList();
+
+
+        List<Dependency> roots0;
+        if (!roots.isEmpty()) {
+            roots0 = resolution.rootDependencies;
+        } else {
+            roots0 = roots;
+        }
+
+        return roots0.stream().<DependencyTree>map(dep -> new Node(dep, false, resolution, withExclusions)).toList();
+    }
```

Comment:
What is the difference between roots and `roots0`? Is there a better name for `roots0`?

Reply:
Will rename

Reply:
This is what roots0 is in the scala code `val roots0 = Option(roots).getOrElse(resolution.rootDependencies)`